lib/Dialect/Torch/Transforms/MaximizeValueSemantics.cpp

```diff
@@ -292,12 +292,12 @@ class RewriteViewLikeSubgraph
     if (copyToValueTensorOps.empty() && viewLikeOps.empty())
       return rewriter.notifyMatchFailure(copy, "no types to change");
 
-    // All uses of `copy` will be updated by the logic below.
-    copy.replaceAllUsesWith(copy.getOperand());
     // All CopyToValueTensorOp operands will be changed to the correct type
     // by the logic below.
     for (CopyToValueTensorOp op : copyToValueTensorOps)
       rewriter.replaceOp(op, op.getOperand());
+    // All uses of `copy` will be updated by the logic below.
+    copy.replaceAllUsesWith(copy.getOperand());
```

Comment:
LGTM as long as at least one of the the existing tests crashes with the new code

Reply:
Some e2e tests crash. I'll add a lit test that also fails

Reply:
Done: https://github.com/llvm/torch-mlir/pull/1684/files#diff-73baa2b9ed878e33c92f5cbe0fa758bdb08b93b1f7c0686feb033a7f48a3969a